src/Hosting/Server.IntegrationTesting/src/Common/TestPortHelper.cs

```diff
@@ -56,5 +56,34 @@ public static int GetNextSSLPort()
                 }
             }
         }
+
+        private const int BasePort = 5001;
+        private const int MaxPort = 8000;
+        private static int NextPort = BasePort;
+
+        // GetNextPort doesn't check for HttpSys urlacls.
+        public static int GetNextHttpSysPort(string scheme)
+        {
+            while (NextPort < MaxPort)
+            {
+                var port = NextPort++;
+
+                using (var server = new HttpListener())
+                {
+                    server.Prefixes.Add($"{scheme}://localhost:{port}/");
+                    try
+                    {
+                        server.Start();
+                        server.Stop();
+                        return port;
+                    }
+                    catch (HttpListenerException)
+                    {
+                    }
+                }
+            }
+            NextPort = BasePort;
+            throw new Exception("Failed to locate a free port.");
```

Comment:
Any idea how long this would take to fail?

Reply:
The individual failures are instant. You could burn through the whole port range in under a minute if it really came to that. It's never come up in HttpSys.